apps/src/templates/teacherNavigation/LessonMaterialsContainer.tsx

```diff
@@ -127,8 +138,22 @@ const LessonMaterialsContainer: React.FC = () => {
           ))}
         </div>
       )}
-    </div>
+    </>
   );
+
+  const renderView = () => {
+    if (!isUnitAssigned) {
+      return (
+        <div className={styles.centerContent}>
+          <NoUnitAssignedView courseName={courseName} />
+        </div>
+      );
+    } else {
+      return defaultView;
+    }
+  };
+
+  return <div>{renderView()}</div>;
```

Comment:
```suggestion
    if (!isUnitAssigned) {
      return (
        <div className={styles.centerContent}>
          <NoUnitAssignedView courseName={courseName} />
        </div>
      );

  return defaultView
```
nit: I don't think we need a different function for this. Another couple questions:
* Do we need to wrap both the `NoUnitAssignedView` and `defaultView` in a div? If not we can replace the `<>` in default view with a `div` and remove the div from the wrapper.
* Do we want to assign `defaultView` or can we just have it as the return value with an early return for the `NoUnitAssignedView`? If we want to keep the `defaultView` value, we should wrap it in a [useMemo](https://react.dev/reference/react/useMemo)